offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/products/ProductsDbConnector.groovy

```diff
@@ -20,223 +21,279 @@ import java.sql.SQLException
+            statement.setInt(1, offerPrimaryId)
+            ResultSet result = statement.executeQuery()
+            while (result.next()) {
+                Product product = rowResultToProduct(SqlExtensions.toRowResult(result))
+                double quantity = result.getDouble("quantity")
+                ProductItem item = new ProductItem(quantity, product)
+                productItems << item
+            }
+        }
+        return productItems
+    }
+
+    @Override
+    Optional<Product> fetch(ProductId productId) throws DatabaseQueryException {
+        Connection connection = provider.connect()
+        String query = Queries.SELECT_ALL_PRODUCTS + " WHERE productId=?"
+        Optional<Product> product = Optional.empty()
+
+        connection.withCloseable {
+            PreparedStatement preparedStatement = it.prepareStatement(query)
+            preparedStatement.setString(1, productId.identifier.toString())
+            ResultSet result = preparedStatement.executeQuery()
+
+            while (result.next()) {
+                product = Optional.of(rowResultToProduct(SqlExtensions.toRowResult(result)))
+            }
+        }
+        return product
+    }
+
+    @Override
+    void store(Product product) throws DatabaseQueryException {
+        Connection connection = provider.connect()
+
+        connection.withCloseable {
+            PreparedStatement preparedStatement = it.prepareStatement(Queries.INSERT_PRODUCT)
+            preparedStatement.setString(1, getProductType(product))
+            preparedStatement.setString(2, product.description)
+            preparedStatement.setString(3, product.productName)
+            preparedStatement.setDouble(4, product.unitPrice)
+            preparedStatement.setString(5, product.unit.value)
+            preparedStatement.setString(6, product.productId.toString())
+
+            preparedStatement.execute()
+        }
+    }
+
+    @Override
+    void archive(Product product) throws DatabaseQueryException {
+        Connection connection = provider.connect()
 
-  }
+        connection.withCloseable {
+            def statement = connection.prepareStatement(Queries.ARCHIVE_PRODUCT)
+            statement.setString(1, product.productId.toString())
+            statement.execute()
+        }
+    }
+
+    /**
+     * Class that encapsulates the available SQL queries.
+     */
+    private static class Queries {
+
+        /**
+         * Query for all available products.
+         */
+        final static String SELECT_ALL_PRODUCTS = "SELECT * FROM product"
+
+        /**
+         * Query for inserting a product.
+         */
+        final static String INSERT_PRODUCT = "INSERT INTO product (category, description, productName, unitPrice, unit, productId) VALUE(?,?,?,?,?,?)"
+
+        /**
+         * Inactivate a product
+         */
+        final static String ARCHIVE_PRODUCT = "UPDATE product SET active = 0 WHERE productId = ?"
+
```

Comment:
These are also newly added

Reply:
Then we should make sure no code is deleted or modified that you do not intend to modify. Has the code changed or is it simply a github diff error?